pkg/controller/direct/sql/mapping.go

```diff
@@ -473,6 +412,73 @@ func SQLInstanceKRMToGCPInstance(in *krm.SQLInstance, refs *SQLInstanceInternalR
 	return out, nil
 }
 
+func InstanceIpConfigurationKRMToGCP(in *krm.InstanceIpConfiguration, refs *SQLInstanceInternalRefs) *api.IpConfiguration {
+	if in == nil {
+		return nil
+	}
+
+	out := &api.IpConfiguration{
+		AllocatedIpRange:                        direct.ValueOf(in.AllocatedIpRange),
+		AuthorizedNetworks:                      InstanceAuthorizedNetworksKRMToGCP(in.AuthorizedNetworks),
+		EnablePrivatePathForGoogleCloudServices: direct.ValueOf(in.EnablePrivatePathForGoogleCloudServices),
+		Ipv4Enabled:                             direct.ValueOf(in.Ipv4Enabled),
+		PscConfig:                               InstancePscConfigKRMToGCP(in.PscConfig),
+		RequireSsl:                              direct.ValueOf(in.RequireSsl),
+		SslMode:                                 direct.ValueOf(in.SslMode),
+	}
+
+	if in.PrivateNetworkRef != nil {
+		out.PrivateNetwork = refs.privateNetwork
+	}
+
+	if in.EnablePrivatePathForGoogleCloudServices != nil {
+		out.ForceSendFields = append(out.ForceSendFields, "EnablePrivatePathForGoogleCloudServices")
+	}
+	if in.Ipv4Enabled != nil {
+		out.ForceSendFields = append(out.ForceSendFields, "Ipv4Enabled")
+	}
+	if in.RequireSsl != nil {
+		out.ForceSendFields = append(out.ForceSendFields, "RequireSsl")
+	}
+
+	return out
+}
+
+func InstanceAuthorizedNetworksKRMToGCP(in []krm.InstanceAuthorizedNetworks) []*api.AclEntry {
+	out := []*api.AclEntry{}
+	for _, net := range in {
+		out = append(out, &api.AclEntry{
+			Kind:           "sql#aclEntry",
+			ExpirationTime: direct.ValueOf(net.ExpirationTime),
+			Name:           direct.ValueOf(net.Name),
+			Value:          net.Value,
+		})
+	}
+	return out
+}
+
+func InstancePscConfigKRMToGCP(in []krm.InstancePscConfig) *api.PscConfig {
+	if len(in) < 1 {
+		return nil
+	}
+
+	// Note:  For some reason, the KRM API allows []InstancePscConfig. However, in the GCP proto there is only
+	// a single *api.PscConfig. I think there is a mistake in the KRM API; it should not allow a list. For
```

Comment:
This seems likely, there are 

You might want to return an error from Reconcile `if len(in) > 1`, we can always _also_ add a CEL rule later.